containous/maesh
# Add watch namespaces option to the controller

## What does this PR do?

This PR:
- Adds the `watchNamespaces` option to the controller. This allow to watch a limited set of resources.
- Adds the `controller.watchNamespaces` option to the helm chart.
- Refactor the IgnoreWrapper and replace it with a ResourceFilter.
- Refactor the PortMapper so we can stop exposing the ServicePort struct.

Fixes #620

### How to test it

* `make test`
* Install Maesh with the `watchNamespaces` option set. And make sure the controller listen only for events coming from the namespace specified.
